src/codegen/doc/mod.rs

```diff
@@ -234,6 +237,59 @@ fn create_object_doc(w: &mut dyn Write, env: &Env, info: &analysis::object::Info
         Ok(())
     })?;
 
+    if has_builder {
+        let builder_ty = TypeStruct::new(SType::Impl, &format!("{}Builder", info.name));
+        let parent_name = builder_ty.name.clone();
+
+        let mut builder_properties: Vec<_> = properties.iter().collect();
+        for parent_info in &info.supertypes {
+            match env.library.type_(parent_info.type_id) {
+                Type::Class(cl) => {
+                    cl.properties
+                        .iter()
+                        .for_each(|p| builder_properties.push(p));
+                }
+                Type::Interface(iface) => {
+                    iface
+                        .properties
+                        .iter()
+                        .for_each(|p| builder_properties.push(p));
+                }
+                _ => (),
+            }
+        }
+        for property in builder_properties.iter() {
+            let ty = TypeStruct {
+                ty: SType::Fn,
+                name: nameutil::signal_to_snake(&property.name),
+                parent: Some(Box::new(builder_ty.clone())),
+                args: vec![],
+            };
+            write_item_doc(w, &ty, |w| {
+                if let Some(ref doc) = property.doc {
+                    writeln!(
+                        w,
+                        "{}",
+                        reformat_doc(&fix_param_names(doc, &None), &symbols, &parent_name)
+                    )?;
+                }
+                if let Some(ver) = property.deprecated_version {
+                    writeln!(w, "\n# Deprecated since {}\n", ver)?;
+                } else if property.doc_deprecated.is_some() {
+                    writeln!(w, "\n# Deprecated\n")?;
+                }
+                if let Some(ref doc) = property.doc_deprecated {
```

Comment:
I'm hoping to remove these soon in favour of `#[deprecated = "msg"]` which, by the way, is not used here or is it emitted elsewhere?

Reply:
I should try it on gtk3 which has more deprecated items

Reply:
The builder pattern doesn't seem to generate a deprecated guard for the generated functions, it doesn't generate anything at all other than the version guard for properties so I guess let's keep this one here until we move to using guard for deprecated elsewhere

Reply:
I'm not sure how to remove them anyway. `#[deprecated]` is generated (at least supposed to) during normal generation and not part of the documentation. Since it can't include the reasoning (LGPL, and that information is simply not available there) only a `#[deprecated = "Since XXX"]` attribute is created.

Besides, putting the full deprecation text in there isn't going to fit nor look pretty.

If you have some time feel free to look into the missing deprecation warning for builders and at least get rid of the `Deprecated`/`Deprecated since XXX` message here (leaving `doc_deprecated` in place). Otherwise I'll enqueue it on my list.

Reply:
Found it, its handling is missing from `fn generate_builder` in `codegen/object.rs`. Will PR later.

Reply:
We really need to come up with an `fn give_me_all_the_attributes_at_once` :sweat_smile:

Reply:
So what I should do here, keep those as is so that you don't forget adding them in your PR? or drop them for now?

Reply:
> We really need to come up with an `fn give_me_all_the_attributes_at_once` sweat_smile

What would such fn do? :)

Reply:
The removal is taken care of in https://github.com/gtk-rs/gir/pull/1148, and the missing Rust annotations were added in https://github.com/gtk-rs/gir/pull/1147. You can probably remove these now.

Reply:
Thanks, done :-)